### ISSUE TITLE
IDL index should have links back to defining section

### ISSUE THREAD

**Issue by tabatkins** [tabatkins/bikeshed#798]
The IDL index collects all the IDL snippets in the spec into one spot, for easy perusal. However, there's no easy way to go back to the definitions when you find something interesting, besides hoping there's an appropriate link you can click.

I should make sure all IDL sections have IDs (generating them if necessary), and associate each run of IDL with its original block, generating a link back to that block in the index.

If I collect anything else (like grammars), same deal.

**Comment by domenic** [comment]
This seems like a great idea. One comment.

> I should make sure all IDL sections have IDs (generating them if necessary)

As with examples, if these could be stable-ish over time that would be ideal. Either by adding a warning mode to make spec authors do it, or by generating them based on the first IDL identifier in the block, or similar. I just want people to be able to Ctrl+C Ctrl+V whatever's in their location bar into a document and have it be valid months later :).

**Comment by tabatkins** [comment]
Yeah, that's the hard one. First identifier probably is more robust than hashing the contents.
